hazelcast-jet-spring/src/main/java/com/hazelcast/jet/spring/JetSpringServices.java

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.spring;
+
+import com.hazelcast.function.BiFunctionEx;
+import com.hazelcast.function.BiPredicateEx;
+import com.hazelcast.function.FunctionEx;
+import com.hazelcast.jet.pipeline.BatchStage;
+import com.hazelcast.jet.pipeline.ServiceFactory;
+import com.hazelcast.jet.pipeline.StreamStage;
+import com.hazelcast.spring.context.SpringAware;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Utility class with methods that create several useful Spring Bean
+ * {@link ServiceFactory service factories} and transform functions.
+ *
+ * @since 4.0
+ */
+public final class JetSpringServices {
+
+    private JetSpringServices() {
+    }
+
+    /**
+     * Returns a transform function for the {@link BatchStage#apply(FunctionEx)}
+     * which adds a mapping stage using the specified bean.
+     *
+     * @param beanName     the name of the bean
+     * @param requiredType the class of the bean
+     * @param mapper       the mapping function
+     * @param <T>          the type of the bean
+     * @param <R>          the type of the returned stage
+     */
+    public static <I, R, T> FunctionEx<BatchStage<I>, BatchStage<R>> mapBatchUsingSpringBean(
```

Comment:
I don't think we need these variants. They don't bring much convenience, and there's too many variations. The plain one using ServiceFactory is good enough and quite convenient. What do you think @jerrinot ?

Reply:
I really like to be able to write: `stage.apply(mapStreamUsingSpringBean("risk-calculator", RiskCalculator::calculateRisk))`

It reads nicely.